File: 2021/java/src/test/java/day01/Day01Test.java
package day01;

import common.Util;
import org.junit.Test;

import java.io.IOException;
import java.net.URISyntaxException;

import static org.junit.Assert.*;

public class Day01Test {

    @Test
    public void pt1test() throws IOException, URISyntaxException {
        Day01 day01 = new Day01();
        String puzzleInput = Util.readFileAsString("day01/test.txt");
        assertEquals(7, day01.pt1(puzzleInput));
    }
    @Test
    public void pt2test() throws IOException, URISyntaxException {
        Day01 day01 = new Day01();
        String puzzleInput = Util.readFileAsString("day01/test.txt");
        assertEquals(5, day01.pt2(puzzleInput));
    }

    @Test
    public void pt1() throws IOException, URISyntaxException {
        Day01 day01 = new Day01();
        String puzzleInput = Util.readFileAsString("day01/input.txt");
        assertEquals(1713, day01.pt1(puzzleInput));
    }
    @Test
    public void pt2() throws IOException, URISyntaxException {
        Day01 day01 = new Day01();
        String puzzleInput = Util.readFileAsString("day01/input.txt");
        assertEquals(1734, day01.pt2(puzzleInput));
    }



    @Test
    public void diff() {
        int[] array1 = {2, 3};
        int[] array2 = {4, 5};
        Day01 day01 = new Day01();
        assertArrayEquals(new int[] {-2, -2}, day01.diff(array1, array2));
    }


    @Test
    public void tail() {
        Day01 day01 = new Day01();
        assertArrayEquals(new int[] {2, 3}, day01.tail(new int[] {1, 2, 3}));
    }
}

File: 2022/scala/src/test/scala/yr2022/days/Day01Test.scala
package yr2022.days

import yr2022.common.InputData
import yr2022.days.Day01.CalorieList

class Day01Test extends org.scalatest.flatspec.AnyFlatSpec {

  it should "do test1" in {
    val inputData = InputData.fromFile("days/day01/test.txt")
    assert(Day01.parseFile(inputData).size == 5)
    assert(Day01.parseFile(inputData) == List(CalorieList(List("1000", "2000", "3000")), CalorieList(List("4000")), CalorieList(List("5000", "6000")), CalorieList(List("7000", "8000", "9000")), CalorieList(List("10000"))))
  }

  it should "max" in {
    val inputData = InputData.fromFile("days/day01/test.txt")
    val m = Day01.pt1(inputData)
    assert(m == 24000)
  }

  it should "pt1" in {
    val inputData = InputData.fromFile("days/day01/input.txt")
    val m = Day01.pt1(inputData)
    assert(m == 71934)
  }

  it should "pt2" in {
    val inputData = InputData.fromFile("days/day01/input.txt")
    val m = Day01.pt2(inputData)
    assert(m == 71934)
  }

}
